File: src/main/java/at/meks/powerdata/PowerData.java
package at.meks.powerdata;

import java.time.Year;
import java.util.List;
import java.util.stream.Stream;

public class PowerData {

    private final List<SinglePowerData> powerData;

    public PowerData(List<SinglePowerData> powerData) {
        this.powerData = powerData;
    }

    public Stream<SinglePowerData> stream(Year year) {
        return powerData.stream()
                        .filter(data -> data.timestampUntil().minusSeconds(1).getYear() == year.getValue());
    }

    public Stream<SinglePowerData> stream() {
        return powerData.stream();
    }
}

File: src/main/kotlin/at/meks/powerdata/PowerData.kt
package at.meks.powerdata

import at.meks.Year
import at.meks.minusSeconds
import io.quarkus.qute.ImmutableList

class PowerData(private val singlePowerData: List<SinglePowerData>) {

    fun powerdataForYear(year: Year) : List<SinglePowerData> {
        return singlePowerData
            .filter { data ->  data.timestampUntil.minusSeconds(1).year == year.getValue()}
    }

    fun getPowerData(): List<SinglePowerData> {
        return ImmutableList.copyOf(singlePowerData)
    }

}
